services/audit-service/src/main/java/datawave/microservice/audit/replay/ReplayController.java

```diff
@@ -0,0 +1,726 @@
+            statusCache.update(status);
+        }
+    }
+    
+    /**
+     * Updates all audit replays
+     *
+     * @param sendRate
+     *            The number of messages to send per second
+     * @return status, indicating the number of audit replays which were successfully updated
+     */
+    @ApiOperation(value = "Updates all audit replays.")
+    @RolesAllowed({"Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/updateAll", method = org.springframework.web.bind.annotation.RequestMethod.POST)
+    public String updateAll(@ApiParam(value = "The number of messages to send per second", required = true) @RequestParam long sendRate,
+                    HttpServletResponse response) {
+        
+        log.info("Updating sendRate to " + sendRate + " for all audit replays");
+        
+        String resp;
+        
+        // only update if the send rate is valid
+        if (sendRate >= 0) {
+            resp = updateAll(sendRate, replayProperties.isPublishEvents(), false) + "audit replays updated";
+        } else {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            resp = "Send rate must be >= 0";
+        }
+        
+        log.info(resp);
+        return resp;
+    }
+    
+    private int updateAll(long sendRate, boolean publishEnabled, boolean onlyUpdateRunning) {
+        int replaysUpdated = 0;
+        
+        if (publishEnabled)
+            appCtx.publishEvent(new AuditReplayRemoteRequestEvent(this, busProperties.getId(), Request.updateAll(sendRate)));
+        
+        // update all of our replays.
+        List<Status> statuses = statusAll();
+        if (onlyUpdateRunning)
+            statuses = statuses.stream().filter(status -> status.getState() == RUNNING).collect(Collectors.toList());
+        
+        for (Status status : statuses) {
+            update(status, sendRate, false);
+            replaysUpdated++;
+        }
+        
+        return replaysUpdated;
+    }
+    
+    /**
+     * Stops an audit replay
+     *
+     * @param id
+     *            The audit replay id
+     * @return status, indicating whether the audit replay was successfully stopped
+     */
+    @ApiOperation(value = "Stops an audit replay.")
+    @RolesAllowed({"Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/{id}/stop", method = org.springframework.web.bind.annotation.RequestMethod.POST)
+    public String stop(@ApiParam("The audit replay id") @PathVariable("id") String id, HttpServletResponse response) {
+        
+        log.info("Stopping audit replay with id " + id);
+        
+        String resp;
+        
+        Status status = status(id, replayProperties.isPublishEvents());
+        if (status != null) {
+            if (stop(status, replayProperties.isPublishEvents())) {
+                resp = "Stopped audit replay with id " + id;
+            } else {
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
same comments as before about the response type

Reply:
If the audit replay is not running, and you try to stop it, is that not a bad request?  Should that be SC_FORBIDDEN or something like that instead?

Reply:
Maybe a 409 fits better. All I know is the syntax of the request is fine, so 400 isn't the right answer.

Reply:
I am leaning towards 422 - UNPROCESSABLE_ENTITY as this implies that the server understands the request, it is syntactically correct, but was unable to process the instructions.  The error message I am sending along with this status code should make it clear what went wrong.